### DISCUSSION TITLE
Why `dev` version seems to be outdated?

### DISCUSSION THREAD

**Discussion by boltlessengineer** [lumen-oss/rocks.nvim#511] [Q&A] [answered]
I'm trying to install latest version of [tree-sitter-http] via rocks.nvim. The `scm-1` version has been [updated few hours ago](https://luarocks.org/modules/neorocks/tree-sitter-http).
But when I install it with `:Rocks install tree-sitter-http dev`, I still get outdated version.
I thought update from [nvim-treesitter] and [nurr] are all steps required to install latest version of treesitter parser.

Are there more steps I have to wait or am I doing something wrong?

[tree-sitter-http]: https://github.com/nvim-rest/tree-sitter-http
[nvim-treesitter]: https://github.com/nvim-treesitter/nvim-treesitter
[nurr]: https://github.com/nvim-neorocks/nurr

**Comment by boltlessengineer** [comment, marked as answer]
Solved: seems like I have to wait until [rocks-binaries-dev](https://nvim-neorocks.github.io/rocks-binaries-dev/) to be updated as well.

**Comment by mrcjkb** [comment]
We have periodic workflows in [the NURR](https://github.com/nvim-neorocks/nurr) which

- Update the parsers' lockfile
- Build the parsers
- If they succeed to build, uploads them to luarocks.org

However, to build the parsers locally, you need a C compiler (available on most linux systems, but not always available on other platforms), `tree-sitter-cli`, and sometimes Node JS (if the parser needs to be generated).
This can be quite a lot, which is why we host pre-built parsers on [our binary servers](https://github.com/nvim-neorocks/rocks-binaries).

That repo has a separate workflow, which

- Installs the parser from luarocks.org
- Packs it (as a `.rock` archive, which is basically a zip archive)
- Updates a manifest that rocks.nvim tells luarocks to prioritise.

This workflow runs every 12 hours.

Since rocks.nvim prioritises the pre-built parsers, they can be a bit outdated (by up to 12 hours).
We have an [open PR](https://github.com/nvim-neorocks/rocks.nvim/pull/494) that will allow you to disable the binary servers.
